core/src/integration-test/java/com/scalar/db/api/DistributedTransactionAdminRepairTableIntegrationTestBase.java

```diff
@@ -0,0 +1,175 @@
+    admin.createCoordinatorTables(true, options);
+  }
+
+  protected Map<String, String> getCreateOptions() {
+    return Collections.emptyMap();
+  }
+
+  private void dropTable() throws ExecutionException {
+    admin.dropTable(getNamespace(), TABLE);
+    admin.dropNamespace(getNamespace());
+    admin.dropCoordinatorTables(true);
+  }
+
+  @BeforeAll
+  protected void beforeAll() throws Exception {}
+
+  @BeforeEach
+  protected void beforeEach() throws Exception {
+    TransactionFactory factory =
+        TransactionFactory.create(TestUtils.addSuffix(getProperties(), TEST_NAME));
+    admin = factory.getTransactionAdmin();
+    createTable();
+    adminTestUtils = AdminTestUtils.create(TestUtils.addSuffix(getStorageProperties(), TEST_NAME));
+  }
+
+  @AfterEach
+  protected void afterEach() throws Exception {
+    dropTable();
+  }
+
+  @AfterAll
+  protected void afterAll() throws Exception {}
+
+  @Test
+  public void repairTableAndCoordinatorTable_ForDeletedMetadataTable_ShouldRepairProperly()
+      throws Exception {
+    // Arrange
+    adminTestUtils.dropMetadataTable();
+
+    // Act
+    admin.repairTable(getNamespace(), getTable(), TABLE_METADATA, getCreateOptions());
+    admin.repairCoordinatorTables(getCreateOptions());
+
+    // Assert
+    assertThat(admin.tableExists(getNamespace(), TABLE)).isTrue();
+    assertThat(admin.getTableMetadata(getNamespace(), TABLE)).isEqualTo(TABLE_METADATA);
+    assertThat(admin.coordinatorTablesExist()).isTrue();
+    assertTableMetadataForCoordinatorTableArePresent();
+  }
+
+  @Test
+  public void repairTableAndCoordinatorTable_ForTruncatedMetadataTable_ShouldRepairProperly()
+      throws Exception {
+    // Arrange
+    adminTestUtils.truncateMetadataTable();
+
+    // Act
+    admin.repairTable(getNamespace(), getTable(), TABLE_METADATA, getCreateOptions());
+    admin.repairCoordinatorTables(getCreateOptions());
+
+    // Assert
+    assertThat(admin.tableExists(getNamespace(), TABLE)).isTrue();
+    assertThat(admin.getTableMetadata(getNamespace(), TABLE)).isEqualTo(TABLE_METADATA);
+    assertTableMetadataForCoordinatorTableArePresent();
+  }
+
+  private void assertTableMetadataForCoordinatorTableArePresent() throws Exception {
+    // There is no coordinator table when using JdbcTransaction
+    if (getStorageProperties().getProperty(DatabaseConfig.TRANSACTION_MANAGER, "").equals("jdbc")) {
+      return;
+    }
+    Properties properties = TestUtils.addSuffix(getStorageProperties(), TEST_NAME);
+    String coordinatorNamespace =
+        new ConsensusCommitConfig(new DatabaseConfig(properties))
+            .getCoordinatorNamespace()
+            .orElse(Coordinator.NAMESPACE);
+    String coordinatorTable = Coordinator.TABLE;
+    // Use the DistributedStorageAdmin instead of the DistributedTransactionAdmin because the latter
+    // expects the
+    // table to hold transactional table metadata columns which is not the case for the coordinator
+    // table
```

Comment:
```suggestion
    // Use the DistributedStorageAdmin instead of the DistributedTransactionAdmin because the latter
    // expects the table to hold transactional table metadata columns which is not the case for the
    // coordinator table
```

Reply:
Thank you, I revised it.